guardrails/formatters/json_formatter.py

```diff
@@ -0,0 +1,127 @@
+import json
+from typing import Optional, Union
+
+from jsonformer import Jsonformer
+
+from guardrails.formatters.base_formatter import BaseFormatter
+from guardrails.llm_providers import (
+    ArbitraryCallable,
+    HuggingFacePipelineCallable,
+    HuggingFaceModelCallable,
+)
+
+
+def _deref_schema_path(schema: dict, path: Union[list, str]):
```

Comment:
Since we're already recursing through the schema it doesn't matter as much, but if you need to deref in the future you should be able to use `jsonref.replace_refs` so you don't have to write it yourself.

Reply:
🫠 Welp...
Might be worth pulling this code, then, just to decrease surface area.  I'll give those a look to see if we can sub them.

Reply:
I've made a note, since there doesn't seem to be a straightforward swap and there are already tests in place for the function.